src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthController.java

```diff
@@ -88,35 +100,54 @@ public String name()
         return OAuthNukleus.NAME;
     }
 
+    public CompletableFuture<Long> resolve(
+        String realmName)
+    {
+        return resolve(realmName, EMPTY_STRING_ARRAY, null);
+    }
+
     public CompletableFuture<Long> resolve(
         String realmName,
-        String... roleNames)
+        String[] roleNames)
     {
-        return resolve(realmName, roleNames, null, null);
+        return resolve(realmName, roleNames, null);
     }
 
     public CompletableFuture<Long> resolve(
         String realmName,
         String[] roleNames,
-        String issuerName,
-        String audienceName)
+        String extension)
     {
+        String issuerName = null;
+        String audienceNames = null;
+        if(extension != null)
+        {
+            final JsonParser parser = new JsonParser();
+            final JsonElement element = parser.parse(extension);
+            if(element.isJsonObject())
+            {
+                final JsonObject object = (JsonObject) element;
+                issuerName = gson.fromJson(object.get(ISSUER), String.class);
+                audienceNames = gson.fromJson(object.get(AUDIENCE), String.class);
```

Comment:
We only support enforcing a single `audience`, so this would be `audienceName` rather than `audienceNames`.